languages/zCEE3.groovy

```diff
@@ -32,80 +32,67 @@ sortedList.each { buildFile ->
-        process.waitFor()
-        if (props.verbose)
-            println("** Exit value for the gradle build: ${process.exitValue()}");
-            
-        // write outputs to log file
-        String enc = props.logEncoding ?: 'IBM-1047'
-        logFile.withWriter(enc) { writer ->
-            writer.append(shellOutput)
-            writer.append(shellError)
-        }
-        
-        if (process.exitValue() != 0) {
-            def errorMsg = "*! Error during the gradle process" 
+        UnixExec zCEE3Execution = new UnixExec().command(shellEnvironment)
+        zCEE3Execution.setOptions(optionsList)
+        zCEE3Execution.output(logFile.getAbsolutePath()).mergeErrors(true);
+        zCEE3Execution.setWorkingDirectory(workingDir)
+        zCEE3Execution.setFile(buildFile)
+        zCEE3Execution.setOutputEncoding(encoding)
+        zCEE3Execution.addOutput(props.buildOutDir, "zCEE3/$WarLocation", "zCEE3")
+        int returnCode = zCEE3Execution.execute()
+
+        if (returnCode != 0) {
+            String errorMsg = "*! Error during the gradle process. Please check the gradle log file at '${logFile.getAbsolutePath()}'."
             println(errorMsg)
-            if (props.verbose)
-                println("*! gradle error message:\n${shellError}")
             props.error = "true"
             buildUtils.updateBuildResult(errorMsg:errorMsg)
         } else {
-            if (props.verbose)
-                println("** gradle output:\n${shellOutput}")
             if (WarFile.exists()) {
                 // Copy api.war to the buildOutDir directory
                 File WarFileTarget = new File(props.buildOutDir + '/zCEE3/' + WarLocation);
                 File WarTargetDir = WarFileTarget.getParentFile();
                 WarTargetDir.mkdirs();
-                Files.copy(WarFile.toPath(), WarFileTarget.toPath(), StandardCopyOption.COPY_ATTRIBUTES);
-            
-                AnyTypeRecord zCEEWARRecord = new AnyTypeRecord("USS_RECORD")
-                zCEEWARRecord.setAttribute("file", buildFile)
-                zCEEWARRecord.setAttribute("label", "z/OS Connect EE OpenAPI 3 YAML definition")
-                zCEEWARRecord.setAttribute("outputs", "[${props.buildOutDir}, zCEE3/$WarLocation, zCEE3]")
-                zCEEWARRecord.setAttribute("command", commandString);
-                BuildReportFactory.getBuildReport().addRecord(zCEEWARRecord)
+                Files.copy(WarFile.toPath(), WarFileTarget.toPath(), StandardCopyOption.COPY_ATTRIBUTES);            
             } else {
-                def errorMsg = "*! Error when searching for the 'api.war' file at location '${WarLocation}'"
+                String errorMsg = "*! Error when searching for the 'api.war' file at location '${WarLocation}'"
                 println(errorMsg)
                 props.error = "true"
                 buildUtils.updateBuildResult(errorMsg:errorMsg)
             }   
         }
     }
 }
+
+def fileExists(String fileLoc){
```

Comment:
Do we really need a method for it? It is only used once and could be incorporated into the main logic.

Reply:
It's also used by `CRB.groovy` I think, so it would make sense to put it in a utility script, so it can be used from several scripts.